### DISCUSSION TITLE
bind key with underscore doesn't work

### DISCUSSION THREAD

**Discussion by jT2a** [alpinejs/alpine#3242] [7. Bugs (archived)]
I am trying to bind two variables to a woocommerce data key attribute.

<a class="add_to_cart_button ajax_add_to_cart hidden" **:data-product_id="slide.id" :data-product_sku="slide.sku"**>Add to Cart</a>

But it doesn't render correctly because of the underscore in the key attribute. Any workarounds for this?

![Screen Shot 2022-11-02 at 12 00 42 pm](https://user-images.githubusercontent.com/97709041/199370483-09d026b2-a8a4-48e5-a93a-fb4a672571c6.png)

**Comment by ekwoka** [comment]
Sure, just use a hyphen.

**Reply by jT2a** [reply]
but that's not the woo commerce api, so it's not going to work.

**Reply by ekwoka** [reply]
Oh right, woocommerce is still a thing...

I'll see if I can identify the issue, it seems like if it's a valid property that you can add with `setAttribute` it should work fine.

**Comment by SergiArias** [comment]
exactly the same problem here. https://github.com/alpinejs/alpine/discussions/3258

**Comment by SergiArias** [comment]
A workaround, although not as simple as x-bind: 
```js
init() {
      $watch('variation_id', value => this.$refs.addtocart.dataset['variation_id'] = value);
}
```

**Reply by SimoTod** [reply]
`x-effect="$el.setAttribute('data-variation_id', variation_id)"`

**Reply by jT2a** [reply]
This is the answer, thank you.

My final code. 

` x-effect="$el.setAttribute('data-product_id', slide.id), $el.setAttribute('data-product_sku', slide.sku)"`
